components/ws-manager-mk2/controllers/timeout_controller.go

```diff
@@ -24,6 +26,9 @@ import (
 )
 
 func NewTimeoutReconciler(c client.Client, cfg config.Configuration, activity *wsactivity.WorkspaceActivity) (*TimeoutReconciler, error) {
+	if cfg.HeartbeatInterval == 0 {
+		return nil, fmt.Errorf("invalid heartbeat interval, must not be 0")
+	}
```

Comment:
additional sanity check: if the heartbeat interval isn't set, the controller should return an error, as otherwise the `reconcileInterval` will be 0 and no timeout checks will get requeued